crd: Add categories for cilium CRDs.

sayboras opened the pull request:
This commit is to add categories for cilium CRDs, so that users can
easily query cilium CRs with kubectl commands (e.g. kubectl get
cilium or kubectl get ciliumpolicy). Two categories (e.g. cilium and
ciliumpolicy) are added to start with.

Signed-off-by: Tam Mach <sayboras@yahoo.com>

Related slack [conversation](https://cilium.slack.com/archives/C2B917YHE/p1628855497104100)

```release-note
crd: Add categories for cilium CRDs
```

sayboras committed:
crd: Add categories for cilium CRDs.

This commit is to add categories for cilium CRDs, so that users can
easily query cilium CRs with kubectl commands (e.g. kubectl get
cilium or kubectl get ciliumpolicy). Two categories (e.g. cilium and
ciliumpolicy) are added to start with.

Signed-off-by: Tam Mach <sayboras@yahoo.com>

nathanjsweet reviewed (approved):
It would be nice if we could add some CI to ensure that future crds do not forget to add the categories field.

sayboras committed:
build(crds): Add check for CRDs

This commit is to add a simple check for generated CRDs to make sure
that all CRDs must have category field specified, also, it must have
cilium category.

Signed-off-by: Tam Mach <sayboras@yahoo.com>

sayboras commented on tools/crdcheck/main.go:
```diff
@@ -0,0 +1,82 @@
+		}
+
+		for _, f := range allChecks {
+			if err = f(crd); err != nil {
```
currently, we have only one check, so i didn't use multi error here. Let's see how it's going.
